File: designPattern/Entry/Basic/src/Entry.java
public abstract class Entry{
    public abstract String getName();
    public abstract int getSize();
    public abstract Entry add(Entry entry);

    public void printList(){
        printList("");
    }
    public abstract void printList(String prefix);

    public String toString(){
        return getName()+" ("+getSize()+")";
    }
}

File: designPattern/Composite/basic2/Entry.py
import abc


class Entry:
    __metaclass__ = abc.ABCMeta

    def get_name(self):
        pass

    def get_size(self):
        pass

    def print_line(self):
        self.print_path("")

    def print_path(self, prefix):
        pass

    def __repr__(self):
        return self.get_name() + " (" + str(self.get_size()) + ")\n"
